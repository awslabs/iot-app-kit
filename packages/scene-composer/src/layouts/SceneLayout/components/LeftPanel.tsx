import React, { FC } from 'react';
import styled from 'styled-components';

import { Direction } from './utils';
import FoldableContainer from './FoldableContainer';
import TabbedPanelContainer from './TabbedPanelContainer';

const FixedWidthDiv = styled.div`
  width: 400px;
  height: 100%;
  overflow-y: auto;
`;

const LeftPanel: FC<Record<string, JSX.Element>> = (props) => (
  <FoldableContainer direction={Direction.Left}>
    <FixedWidthDiv>
      <TabbedPanelContainer panels={props} />
    </FixedWidthDiv>
  </FoldableContainer>
);

export default LeftPanel;
