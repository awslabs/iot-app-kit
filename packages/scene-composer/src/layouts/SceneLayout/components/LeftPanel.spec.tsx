import React from 'react';
import renderer from 'react-test-renderer';

import LeftPanel from './LeftPanel';

jest.mock('./FoldableContainer', () => (props) => <div data-testid='FoldableContainer' {...props} />);

jest.mock('./TabbedPanelContainer', () => (props) => <div data-testid='TabbedPanelContainer' {...props} />);

describe('<LeftPanel />', () => {
  it('should render the expencted layout', () => {
    const container = renderer.create(<LeftPanel TestPanel={<div>TestPanel</div>} />);

    expect(container).toMatchSnapshot();
  });
});
