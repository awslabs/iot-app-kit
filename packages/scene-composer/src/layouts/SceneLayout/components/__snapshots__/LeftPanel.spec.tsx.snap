// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftPanel /> should render the expencted layout 1`] = `
.c0 {
  width: 400px;
  height: 100%;
  overflow-y: auto;
}

<div
  data-testid="FoldableContainer"
  direction="Left"
>
  <div
    className="c0"
  >
    <div
      data-testid="TabbedPanelContainer"
      panels={
        Object {
          "TestPanel": <div>
            TestPanel
          </div>,
        }
      }
    />
  </div>
</div>
`;
